slicebase/etc/mlab/slicectrl-functions

```diff
@@ -45,12 +46,12 @@ function stop_after_everything () {
 function kill_remaining () {
     local signal=$1
     KILLED=
-    pgrep -v -f "slicectrl|rsync|rpm" > /var/run/$SLICENAME/pids.afterstop
+    pgrep -v -f "slicectrl|rsync|rpm|bash|login|/bin/su" > /var/run/$SLICENAME/pids.afterstop
     if [ -f /var/run/$SLICENAME/pids.beforestart ] ; then
         for pid in `comm -23 <( sort /var/run/$SLICENAME/pids.afterstop ) \
-                             <( sort /var/run/$SLICENAME/pids.beforestart )`; do 
-            echo "STILL RUNNING: " `ps ax | grep $pid`
-            kill -$signal $pid
+                             <( sort /var/run/$SLICENAME/pids.beforestart )`; do
+            echo "STILL RUNNING: " `ps ax | grep $pid | grep -v grep` 1>&2
+            kill -$signal $pid 2> /dev/null
             KILLED="TRUE"
```

Comment:
Very good idea.

Another situation I saw was that we would try to send TERM to a process that existed when the `pgrep` ran, but not when `kill -TERM` ran.

There was a bug here since the second phase of `kill_remaining` (using sigkill) was triggered on whether KILLED was set. Previously it was set unconditionally, which included cases even when the process no longer existed. Which was incorrect.